shell/platform/windows/flutter_window.h

```diff
@@ -9,163 +9,332 @@
 
   // |FlutterWindowBindingHandler|
-  void SetFlutterCursor(HCURSOR cursor) override;
+  virtual void SetFlutterCursor(HCURSOR cursor) override;
 
   // |FlutterWindowBindingHandler|
-  void OnWindowResized() override;
+  virtual void OnWindowResized() override;
 
   // |FlutterWindowBindingHandler|
-  bool OnBitmapSurfaceUpdated(const void* allocation,
-                              size_t row_bytes,
-                              size_t height) override;
+  virtual bool OnBitmapSurfaceUpdated(const void* allocation,
+                                      size_t row_bytes,
+                                      size_t height) override;
 
   // |FlutterWindowBindingHandler|
-  PointerLocation GetPrimaryPointerLocation() override;
+  virtual PointerLocation GetPrimaryPointerLocation() override;
 
-  // |Window|
-  void OnThemeChange() override;
+  // Called when a theme change message is issued.
+  virtual void OnThemeChange();
 
   // |WindowBindingHandler|
-  void SendInitialAccessibilityFeatures() override;
+  virtual void SendInitialAccessibilityFeatures() override;
 
   // |WindowBindingHandler|
-  AlertPlatformNodeDelegate* GetAlertDelegate() override;
+  virtual AlertPlatformNodeDelegate* GetAlertDelegate() override;
 
   // |WindowBindingHandler|
-  ui::AXPlatformNodeWin* GetAlert() override;
+  virtual ui::AXPlatformNodeWin* GetAlert() override;
 
   // |WindowBindingHandler|
-  bool NeedsVSync() override;
+  virtual bool NeedsVSync() override;
+
+  // Called to obtain a pointer to the fragment root delegate.
+  virtual ui::AXFragmentRootDelegateWin* GetAxFragmentRootDelegate();
+
+  // Called on a resize or focus event.
+  virtual void OnWindowStateEvent(WindowStateEvent event);
+
+ protected:
+  // OS callback called by message pump.  Handles the WM_NCCREATE message which
+  // is passed when the non-client area is being created and enables automatic
+  // non-client DPI scaling so that the non-client area automatically
+  // responsponds to changes in DPI.  All other messages are handled by
+  // MessageHandler.
+  static LRESULT CALLBACK WndProc(HWND const window,
+                                  UINT const message,
+                                  WPARAM const wparam,
+                                  LPARAM const lparam) noexcept;
+
+  // Win32's DefWindowProc.
+  //
+  // Used as the fallback behavior of HandleMessage. Exposed for dependency
+  // injection.
+  virtual LRESULT Win32DefWindowProc(HWND hWnd,
+                                     UINT Msg,
+                                     WPARAM wParam,
+                                     LPARAM lParam);
+  // Release OS resources associated with the window.
+  void Destroy();
+
+  // Converts a c string to a wide unicode string.
+  std::wstring NarrowToWide(const char* source);
+
+  // Registers a window class with default style attributes, cursor and
+  // icon.
+  WNDCLASS RegisterWindowClass(std::wstring& title);
```

Comment:
Could you do a pass on `protected` members to check if they can be made `private` instead? It looks like `RegisterWindowClass` should be private, there may be more.